### PR TITLE
TST: wait for conditions during troublesome tests

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Try to fix tests in #169.  Changes like these could be spread in many places

Basically just assert things with a timeout to give conditions the ability to settle.

## Motivation and Context
#169 , and other tests

## How Has This Been Tested?
Locally, and by hitting the GHA re-run button a bunch of times, likely

## Where Has This Been Documented?
This PR.

### CODE REVIEW

**Comment by tangkong** [comment]
I can expand this treatment to other tests, but I also haven't managed to see a test suite failure throughout the course of this PR.  If anyone has a better stress test methodology I'd love to try it

**Comment by ZLLentz** [comment]
One way to stress test this is to use pytest-repeat but that might reveal other test suite issues
```
pytest --count=1000 tests/bad_tests.py
```

File: lightpath/tests/test_widgets.py [hunk 1]
```diff
@@ -27,15 +28,32 @@ def test_widget_updates(lightrow: LightRow, path: BeamPath):
...
     ipimb.insert()
-    assert (to_stylesheet_color(state_colors['half_removed'])
-            in ipimb_row.state_label.styleSheet())
+
+    def half_removed():
+        return (to_stylesheet_color(state_colors['half_removed'])
+                in ipimb_row.state_label.styleSheet())
+
+    wait_until(half_removed)
+    assert half_removed
```
Comment (ZLLentz) [inline lightpath/tests/test_widgets.py:37]: I think these are all typos, you meant to assert the function result, right?
```suggestion
    assert half_removed()
```

File: lightpath/tests/test_gui.py [hunk 2]
```diff
@@ -68,15 +69,15 @@ def test_upstream_check(lightapp: LightApp, monkeypatch):
...
+def test_filtering(qtbot: QtBot, lightapp: LightApp, monkeypatch):
     lightapp.destination_combo.setCurrentIndex(4)  # set current to MEC
     # Create mock functions
     for row in lightapp.rows:
         monkeypatch.setattr(row[0], 'setHidden', Mock())
     # Initialize properly with nothing hidden
     lightapp.filter()
     for row in lightapp.rows:
-        row[0].setHidden.assert_called_with(False)
+        qtbot.waitUntil(lambda: row[0].setHidden.assert_called_with(False))
```
Comment (ZLLentz) [inline lightpath/tests/test_gui.py:80]: How do these statements work? The encapsuled `assert_called_with` is actually an assert statement. Does it have some undocumented return value?
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with

Comment (ZLLentz) [reply lightpath/tests/test_gui.py:80]: Nevermind, I didn't know qtbot would wait until the assertion passed

File: lightpath/tests/test_widgets.py (hunk 1 above)
Comment (tangkong) [reply lightpath/tests/test_widgets.py:37]: hah no wonder it never failed 🤦

File: lightpath/tests/test_gui.py (hunk 2 above)
Comment (tangkong) [reply lightpath/tests/test_gui.py:80]: Yea this is some fanciness I picked up while filling out the atef test suite.

File: lightpath/tests/test_widgets.py (hunk 1 above)
Comment (tangkong) [reply lightpath/tests/test_widgets.py]: Interestingly, locally the `removed` case can seemingly hang forever (10+ seconds, which is effectively forever for a computer brain)

Will investigate further

**Review by ZLLentz — APPROVED** [review]
I'll put my approval in because this seems to help

**Pull request merged**